### PR TITLE
[submodule] update libfmt to tag 9.1.0

### PR DESCRIPTION
**Pull request by cyyever**
cc @ezyang @gchanan

### CODE REVIEW

**Comment by Skylion007** [comment]
@cyyever I opened a PR on libkineto to fix at least one build issue over there: https://github.com/pytorch/kineto/pull/721

**Comment by cyyever** [comment]
@Skylion007 It is better to wait your PR get merged first then I rebase

**Comment by Skylion007** [comment]
Yeah, I am waiting on a review for that repo. @dagitses @malfet not part of the PyTorch Slack yet, would you mind pinging someone on the Kineto to merge the PR. It is a fairly straightforward missing include fix.

**Review by malfet — APPROVED** [review]

**Comment by cyyever** [comment]
@Skylion007  Bazel build failure was triggered, possibly due to bad support on libfmt, and another on ittapi. May be we can ignore that and wait for Bazel maintainers to fix latter. I don't know much about Bazel, can't help.

**Comment by Skylion007** [comment]
@cyyever PyTorch's bazel project file for fmtlib is right here: https://github.com/pytorch/pytorch/blob/master/third_party/fmt.BUILD

File: torch/csrc/Exceptions.cpp
Comment (Skylion007) [inline torch/csrc/Exceptions.cpp:6]: include no longer needed, right?

Comment (cyyever) [reply torch/csrc/Exceptions.cpp:6]: Can remove

**Comment by cyyever** [comment]
@Skylion007 The only test failure was [backwards_compat, 1, 1, linux.2xlarge)], by checking the error, I found the code was built with the old version. I have no permission to fix build script hacks (maybe that is intended). But you may be able to merge this PR by force.

**Comment by Skylion007** [comment]
I'd like to have a bazel expert look here to make sure that the workaround is correct, and also to someone to figure out what to do about this failing test. @malfet @albanD ?

File: .ci/pytorch/build.sh
Comment (albanD) [inline .ci/pytorch/build.sh:200]: @vors does that sound ok to you?

**Review by albanD — COMMENTED** [review]
Is the kineto hash still in the custom branch on back to main.
I don't think we want to point to a special branch as these hash get auto-updated.

**Comment by Skylion007** [comment]
It's been merged to main

File: .ci/pytorch/build.sh
Comment (vors) [reply .ci/pytorch/build.sh:200]: can you try this pattern?

https://github.com/pytorch/pytorch/blob/e52786f3d177a7ca5d490a516cf52e236ef072cb/WORKSPACE#L246-L254

Comment (albanD) [reply .ci/pytorch/build.sh:200]: cc @cyyever can you try that? :)

**Comment by cyyever** [comment]
@albanD  @Skylion007  bazel works now. But [linux-focal-py3.8-gcc7 / test (backwards_compat, 1, 1, linux.2xlarge)] still run test on old code.

File: WORKSPACE
Comment (vors) [inline WORKSPACE:262]: nit: to follow other naming conventions from this section, use `unused_ftm_bazel` as a name.

**Comment by Skylion007** [comment]
@malfet Is there a special tag we need to add for bc_breaking changes? Or do we just force merge it?

**Comment by ezyang** [comment]
No, the job just seems to be straight up broken by the upgrade. I forgot how the BC test works. Does the error message ring ab ell?

**Comment by albanD** [comment]
It is indeed a bug in the bc test. https://github.com/pytorch/pytorch/pull/94182 should fix that.

**Comment by malfet** [comment]
> @malfet Is there a special tag we need to add for bc_breaking changes? Or do we just force merge it?

No, I guess you can safely ignore it, as it will not be visible on trunk. But yes, https://github.com/pytorch/pytorch/pull/94182 should fix it

**Review by Skylion007 — APPROVED** [review]

**Comment by Skylion007** [comment]
FYI, @malfet the fix didn't work.

**Comment by albanD** [comment]
@Skylion007 The logs show:
```
+ git reset --hard 819990f5956e040ab1e4a0c34f205f1691b43977
HEAD is now at 819990f595 [decomp] Decompose std/std_mean into aten.var/var_mean (#94072)
+ echo '::group::Installing Torch From Base Commit'
```

So the change from the other PR is not included here.

**Comment by Skylion007** [comment]
> @Skylion007 The logs show:
> 
> ```
> + git reset --hard 819990f5956e040ab1e4a0c34f205f1691b43977
> HEAD is now at 819990f595 [decomp] Decompose std/std_mean into aten.var/var_mean (#94072)
> + echo '::group::Installing Torch From Base Commit'
> ```
> 
> So the change from the other PR is not included here.

Hmm, maybe I rebased too soon.

**Comment by albanD** [comment]
btw you have a `-r` option for the merge command that chains a rebase and merge.

**Comment by Skylion007** [comment]
> @Skylion007 The logs show:
> 
> ```
> + git reset --hard 819990f5956e040ab1e4a0c34f205f1691b43977
> HEAD is now at 819990f595 [decomp] Decompose std/std_mean into aten.var/var_mean (#94072)
> + echo '::group::Installing Torch From Base Commit'
> ```
> 
> So the change from the other PR is not included here.

Still failing after the rebase... hmm...

**Comment by albanD** [comment]
Yeah not sure why tbh...
If you really want this in, you can force merge it otherwise I would wait for one nightly and try and rebase tomorrow.

**Review by albanD — APPROVED** [review]

**Pull request closed without merging**
